libraries/botbuilder-dialogs-adaptive/src/recognizers/adaptiveRecognizer.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { BoolExpression } from "adaptive-expressions";
+import { getTopScoringIntent, RecognizerResult } from "botbuilder-core";
+import { DialogContext, Recognizer } from "botbuilder-dialogs";
+
+// TODO
+// (differs from C# in that it uses <name-of-recognizer>Configuration interfaces)
+// AdaptiveRecognizerConfiguration interface
+// with optional BoolExpression logPersonalInformation
+
+export abstract class AdaptiveRecognizer extends Recognizer {
+    /**
+     * (Optional) Flag that designates whether personally identifiable information (PII) should log to telemetry.
+     */
+    public logPersonalInformation = new BoolExpression('=settings.telemetry.logPersonalInformation');
+
+    /**
+     * Uses the RecognizerResult to create a list of propeties to be included when tracking the result in telemetry.
+     * @param recognizerResult Recognizer Result.
+     * @param telemetryProperties A list of properties to append or override the properties created using the RecognizerResult.
+     * @param dialogContext Dialog Context.
+     * @returns A dictionary that can be included when calling the TrackEvent method on the TelemetryClient.
+     */
+    protected fillRecognizerResultTelemetryProperties(
+        recognizerResult: RecognizerResult,
+        telemetryProperties: { [key: string]: string },
+        dialogContext?: DialogContext
+    ): { [key: string]: string } {
+        if (dialogContext == null)
```

Comment:
`if (!dialogContext)` should suffice. Also, can you make sure to run code formatting? In vscode, the `prettier` extension is useful for this.

Reply:
Also - if this is indeed a required parameter, why make it optional?

Reply:
- whoops, the perils of switching between C# and JS 😅 
- and I left it as optional and filed it away as "figure out later", since it was optional in `Recognizer` base, and was also unsure of the nuances for binary compat in TS/JS

Changed it locally to be required, and it built...running pipeline, so I guess I'll see now how things go :)